app/javascript/article-form/articleForm.jsx

```diff
@@ -396,6 +399,7 @@ export class ArticleForm extends Component {
           />
         ) : (
           <Form
+            key={formKey}
```

Comment:
I revisited this and changing the [key](https://reactjs.org/docs/lists-and-keys.html#keys) prop appears to be the cleanest solution to this. Doing this prevents us from changing the behavior of the `<MentionAutocompleteTextArea />` component.

When the key prop changes, it causes the `<Form />` component to recreate the DOM nodes that it manages.

Reply:
Nice! Yeah that's a great idea and keeps everything as straightforward as possible 🙌